azure/services/agentpools/spec.go

```diff
@@ -198,29 +198,26 @@ func (s *AgentPoolSpec) Parameters(ctx context.Context, existingObj genruntime.M
 	defer done()
 
 	// If existing is preview, convert to stable then back to preview at the end of the function.
-	var existing *asocontainerservicev1.ManagedClustersAgentPool
-	var existingStatus asocontainerservicev1preview.ManagedClusters_AgentPool_STATUS
+	var existing *asocontainerservicev1hub.ManagedClustersAgentPool
 	if existingObj != nil {
+		hub := &asocontainerservicev1hub.ManagedClustersAgentPool{}
 		if s.Preview {
 			existingPreview := existingObj.(*asocontainerservicev1preview.ManagedClustersAgentPool)
-			existingStatus = existingPreview.Status
-			hub := &asocontainerservicev1hub.ManagedClustersAgentPool{}
 			if err := existingPreview.ConvertTo(hub); err != nil {
 				return nil, err
 			}
-			stable := &asocontainerservicev1.ManagedClustersAgentPool{}
-			if err := stable.ConvertFrom(hub); err != nil {
+		} else {
+			existingStable := existingObj.(*asocontainerservicev1.ManagedClustersAgentPool)
+			if err := existingStable.ConvertTo(hub); err != nil {
 				return nil, err
 			}
-			existing = stable
-		} else {
-			existing = existingObj.(*asocontainerservicev1.ManagedClustersAgentPool)
 		}
+		existing = hub.DeepCopy()
```

Comment:
Why the deep copy here?

Reply:
I think this was left over from a fix I tried to do for the mismatched Status. It definitely isn't needed!